### PR TITLE
En 5910/key value storage for accounts

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: process/smartContract/builtInFunctions/keyValueStorage.go [hunk 1]
```diff
@@ -1,34 +1,75 @@
...
+
+	useGas += k.gasConfig.StorePerByte * lengthChange
+	if input.GasProvided < useGas {
+		return nil, input.GasProvided, process.ErrNotEnoughGas
+	}
 
-// GasUsed -
-func (k *keyValueStorage) GasUsed() uint64 {
-	return 0
+	return big.NewInt(0), useGas, nil
```
Comment (andreibancioiu) [inline process/smartContract/builtInFunctions/keyValueStorage.go:69]: Save key/value to trie.

File: process/smartContract/builtInFunctions/factory.go [hunk 2]
```diff
@@ -7,13 +7,10 @@ import (
...
 )
 
-// claimDeveloperRewardsFunctionName is a constant which defines the name for the claim developer rewards function
 const claimDeveloperRewardsFunctionName = "ClaimDeveloperRewards"
-
-// changeOwnerAddressFunctionName is a constant which defines the name for the change owner address function
 const changeOwnerAddressFunctionName = "ChangeOwnerAddress"
-
 const setUserName = "SetUserName"
+const saveKeyValue = "SaveKeyValue"
```
Comment (raduchis) [inline process/smartContract/builtInFunctions/factory.go:13]: the constants could be saved in core and be also used in the tests

File: process/smartContract/builtInFunctions/keyValueStorage.go (hunk 1 above)
Comment (raduchis) [reply process/smartContract/builtInFunctions/keyValueStorage.go:69]: +1

File: process/smartContract/builtInFunctions/keyValueStorage_test.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+			ReleasePerByte:  1,
+			DataCopyPerByte: 1,
+			PersistPerByte:  1,
+			CompilePerByte:  1,
+		},
+	}
+
+	addr := []byte("addr")
+
+	acc, _ := state.NewUserAccount(mock.NewAddressMock(addr))
```
Comment (raduchis) [inline process/smartContract/builtInFunctions/keyValueStorage_test.go:30]: Add a test to see that the value was stored in the account.

File: cmd/node/factory/structs.go [hunk 4]
```diff
@@ -1727,7 +1727,13 @@ func newShardBlockProcessor(
 		return nil, err
 	}
 
-	txTypeHandler, err := coordinator.NewTxTypeHandler(stateComponents.AddressConverter, shardCoordinator, stateComponents.AccountsAdapter)
+	argsTxTypeHandler := coordinator.ArgNewTxTypeHandler{
+		AdrConv:          stateComponents.AddressConverter,
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply cmd/node/factory/structs.go:1731]: done

File: process/smartContract/builtInFunctions/factory.go (hunk 2 above)
Comment (sasurobert) [reply process/smartContract/builtInFunctions/factory.go:13]: constants will come from config file - andrei is working on it.

File: process/smartContract/builtInFunctions/keyValueStorage.go [hunk 5]
```diff
@@ -1,34 +1,75 @@
...
 
+	"github.com/ElrondNetwork/elrond-go/core/check"
 	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/process"
 	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
 )
 
-type keyValueStorage struct {
-	gasConfig BaseOperationCost
+type saveKeyValueStorage struct {
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/smartContract/builtInFunctions/keyValueStorage.go:14]: I prefer shorter names. would leave like this.

File: process/smartContract/builtInFunctions/keyValueStorage.go (hunk 1 above)
Comment (sasurobert) [reply process/smartContract/builtInFunctions/keyValueStorage.go:69]: done. good catch :)

File: process/smartContract/builtInFunctions/keyValueStorage_test.go (hunk 3 above)
Comment (sasurobert) [reply process/smartContract/builtInFunctions/keyValueStorage_test.go:30]: added.

File: process/smartContract/builtInFunctions/factory.go (hunk 2 above)
Comment (raduchis) [reply process/smartContract/builtInFunctions/factory.go:13]: ok

**Pull request merged**